### DISCUSSION TITLE
Pushing for implementation of cubic interpolation for 3D images.

### DISCUSSION THREAD

**Discussion by Tarek-Alkabbani** [Project-MONAI/MONAI#7806] [Ideas]
I was working on a project of mine that involves 3D MRI scans. These scans had high spatial resolution and would have benefited from using cubic spline interpolation for any transform that would require interpolation in any shape or form. However, when it comes to interpolation, MONAI relies on PyTorch's interpolate function. Sadly, PyTorch supports cubic interpolation for 2D images strictly, and 3D images are usually interpolated using the trilinear mode. 

As there is much more use of 3D images in medical imaging than other pieces of media, I was thinking that it would make sense that this community would care the most for such a change. I wanted to gauge if this idea is popular enough to be pushed into implementation in Pytorch.

Something similar has occurred before with PyTorch's [grid-sample](https://pytorch.org/docs/stable/nn.functional.html#grid-sample) model layer. And higher order interpolation was then implemented #789 . This however has not carried over to transforms.

**Comment by KumoLiu** [comment]
Hi @Tarek-Alkabbani, does this meet your requirement?
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/spatial/array.py#L143
